### PR TITLE
refactor: code splitting to support third party sso/oidc in EE

### LINKED ISSUE #10199: Refactor/Code Split for login pages

### PR DESCRIPTION
**Pull request by pranavkanade**
## Description
Code splitting changes to support sso in ee

This change follows guidelines detailed in this doc - https://www.notion.so/appsmith/FE-code-splitting-guidelines-V3-7333eaa05e1540b898feb540b5f0f330

## Summary
1. UserAuth pages had to be spit to support EE feature. Hence, it's been moved to `/ce/pages/UserAuth` and a corresponding file is also added to `/ee/pages/UserAuth`.
2. Next step is to split the `configs` to as EE feature will deviate from the existing CE version.
3. Declaration of `window` interface had to be moved to as it's dependency `INJECTED_CONFIGS` would also slightly deviate in EE.
4. Fix the imports - change the imports from `pages/UserAuth`, `constants`, and `configs` to `@appsmith/...`
5. Updated the `babelrc`, `jest config` and `tsconfig`

Fixes #10199 

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- Existing automated test cases

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: refactor/login-page 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.12 **(0)** | 36.65 **(0.01)** | 34.54 **(0)** | 55.65 **(0)**
 :sparkles: :new: | **app/client/src/ce/configs/index.ts** | **95.24** | **86.41** | **100** | **97.3**
 :sparkles: :new: | **app/client/src/ce/constants/ApiConstants.tsx** | **96.43** | **100** | **75** | **96.3**
 :red_circle: | app/client/src/components/ads/Icon.tsx | 52.46 **(-0.15)** | 35.54 **(-0.22)** | 100 **(0)** | 52.35 **(-0.14)**
 :sparkles: :new: | **app/client/src/ee/configs/index.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/ee/constants/ApiConstants.tsx** | **100** | **100** | **100** | **100**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**
 :x: | ~~app/client/src/configs/index.ts~~ | ~~95.12~~ | ~~86.41~~ | ~~100~~ | ~~97.3~~
 :x: | ~~app/client/src/constants/ApiConstants.tsx~~ | ~~96.43~~ | ~~100~~ | ~~75~~ | ~~96.3~~</details>

### CODE REVIEW

File: app/client/src/ee/configs/index.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import { INJECTED_CONFIGS } from "ce/configs/index";
+
+declare global {
+  interface Window {
+    APPSMITH_FEATURE_CONFIGS: INJECTED_CONFIGS;
+    Intercom: any;
+    evaluationVersion: EvaluationVersion;
+    Sentry: any;
+  }
+}
```
Comment (rishabhsaxena) [inline app/client/src/ee/configs/index.ts:12]: QQ Was wondering the use-case of having anything else apart from the direct imports and exports from the common or `ce` modules? So here, for instance, why have we moved the `Window` interface to `ee`. I guess we do something similar at couple of other places

**Review by rishabhsaxena — APPROVED** [review]
LGTM, just a quick question
Also had left a comment on the doc for including a minimal example with exact steps, think that would be useful

File: app/client/src/ee/configs/index.ts (hunk 1 above)
Comment (pranavkanade) [reply app/client/src/ee/configs/index.ts:12]: I'll add a walkthrough demo video and upload it there.

Comment (pranavkanade) [reply app/client/src/ee/configs/index.ts:12]: Regarding the Window interface, the dependency `INJECTED_CONFIGS` will be updated in the EE repo to accommodate, keycloak config var, and we'll do that by extending the existing object in `ee` file, hence it had to be moved from ce to ee

**Pull request merged**